docs/how-to-guides/data-basics/4-building-a-create-page-form.md

```diff
@@ -0,0 +1,373 @@
+			<div className="list-controls">
+				<SearchControl onChange={ setSearchTerm } value={ searchTerm } />
+				<PageCreateButton />
+			</div>
+			<PagesList hasResolved={ hasResolved } pages={ pages } />
+		</div>
+	);
+}
+```
+
+The final result should look as follows:
+
+![](/docs/how-to-guides/data-basics/media/create-form/create-button.png)
+
+### Step 2: Extract a controlled PageForm
+
+Now that the button is in place, we may focus entirely on building the form. This tutorial is about managing data, so we will not build a complete page editor. Instead, the form will only contain one field: post title.
+
+Luckily, the `EditPageForm` we have built in [part three](/docs/how-to-guides/data-basics/3-building-an-edit-form.md) already takes us 80% there. The bulk of the user interface is already there, and we will reuse it in the `CreatePageForm`. Let’s start by extracting the form UI into a separate component:
+
+```js
+export function EditPageForm( { pageId, onCancel, onSaveFinished } ) {
+	// ...
+	return (
+		<PageForm
+			title={ page.title }
+			onChangeTitle={ handleChange }
+			hasEdits={ hasEdits }
+			lastError={ lastError }
+			isSaving={ isSaving }
+			onCancel={ onCancel }
+			onSave={ handleSave }
+		/>
+	);
+}
+
+export function PageForm( { title, onChangeTitle, hasEdits, lastError, isSaving, onCancel, onSave } ) {
+	return (
+		<div className="my-gutenberg-form">
+			<TextControl
+				label="Page title:"
+				value={ title }
+				onChange={ onChangeTitle }
+			/>
+			{ lastError ? (
+				<div className="form-error">Error: { lastError.message }</div>
+			) : (
+				false
+			) }
+			<div className="form-buttons">
+				<Button
+					onClick={ onSave }
+					variant="primary"
+					disabled={ ! hasEdits || isSaving }
+				>
+					{ isSaving ? <Spinner /> : 'Save' }
+				</Button>
+				<Button onClick={ onCancel } variant="tertiary">
+					Cancel
+				</Button>
+			</div>
+		</div>
+	);
+}
+```
+
+This code quality change should not change anything about how the application works. Let’s try to edit a page just to be sure:
+
+![](/docs/how-to-guides/data-basics/media/create-form/edit-page-form.png)
+
+Great! The edit form is still there, and now we have a building block to power the new `CreatePageForm`.
+
+### Step 3: Build a CreatePageForm
+
+The only thing that `CreatePageForm` component must do is to provide the following seven properties needed to render the `PageForm` component:
+
+* title
+* onChangeTitle
+* hasEdits
+* lastError
+* isSaving
+* onCancel
+* onSave
+
+Let’s see how we can do that:
+
+#### Title, onChangeTitle, hasEdits
+
+The `EditPageForm` updated and saved an existing entity record that lived in the Redux state. Because of that, we relied on the `editedEntityRecords` selector.
+
+In case of the `CreatePageForm`, however, there is no pre-existing entity record. There is only an empty form. Anything that the user types is local to that form, which means we can kep track of it using the React’s `useState` hook:
```

Comment:
```suggestion
In case of the `CreatePageForm` however, there is no pre-existing entity record. There is only an empty form. Anything that the user types is local to that form, which means we can keep track of it using the React’s `useState` hook:
```